### PR TITLE
MVP implementation

### PR DESCRIPTION
**Pull request by SofiaZarzabal**
- Added ParkingModel, ParkingView, ParkingPresenter classes and ParkingContract interface on packages.

- Added a string in string resources for the Toast's message.

- Implemented the button's functionality.

UICapture

folderStructure

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.example.parkingsystem.mvp.model;
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java:2]: remove this extra space pls

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.example.parkingsystem.mvp.presenter;
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java:2]: remove this extra space pls

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.example.parkingsystem.mvp.contract.ParkingContract;
+import com.example.parkingsystem.mvp.model.ParkingModel;
+import com.example.parkingsystem.mvp.view.ParkingView;
+
+public class ParkingPresenter implements ParkingContract.ParkingPresenter {
+
+    private ParkingModel model;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: here you should have the contract types

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    private ParkingView view;
+
+
+    public ParkingPresenter(ParkingModel model, ParkingView view) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: here you should have the contract types too

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        this.model=model;
+        this.view=view;
+    }
+
+
+    public void onButtonMainSelectParkingPressed() {
+        model.setParkingLots(10);
+        view.showParkingSpaces(model.getParkingLots());
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: remove this extra space pls

File: app/src/main/java/com/example/parkingsystem/mvp/view/ActivityView.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.example.parkingsystem.mvp.view;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ActivityView.java]: this class should be in a package "base" inside the "view" package

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    private final ActivityMainBinding binding;
+
+    public ParkingView(Activity activity, ActivityMainBinding binding) {
+        super(activity);
+        this.binding = binding;
+    }
+
+    public void showParkingSpaces(int parkingLots) {
+        Context context= getContext();
+        Toast.makeText(context, context.getResources().getString(R.string.toast_main_activity_total_parking_lots) + parkingLots, Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: before use the context you have to validate that it is not null here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java (hunk 1 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/model/ParkingModel.java:2]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 2 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java:2]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java (hunk 7 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 5 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 4 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 3 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done!

File: app/src/main/java/com/example/parkingsystem/mvp/view/ActivityView.java (hunk 6 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ActivityView.java]: Done!

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    private ParkingContract.ParkingModel model;
+    private ParkingContract.ParkingView view;
+
+    public ParkingPresenter(ParkingContract.ParkingModel model, ParkingContract.ParkingView view) {
+        this.model = model;
+        this.view = view;
+    }
+
+    public void onButtonMainSelectParkingPressed() {
+        model.setParkingLots(10);
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java:16]: pleas avoid harcodes

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -2,4 +2,5 @@
     <string name="app_name">ParkingSystem</string>
     <string name="text_main_activity_title">Welcome to Parking System</string>
     <string name="button_main_activity_total_parking_lots">Select parking lots</string>
+    <string name="toast_main_activity_total_parking_lots">Total of parking lots: </string>
```
Comment (nicolbidone) [inline app/src/main/res/values/strings.xml]: here the resource should be 
Total of parking lots:  %1$d
so you can use
context.getString(R.string.toast_main_activity_total_parking_lots, parkingLots)

Comment (SofiaZarzabal) [reply app/src/main/res/values/strings.xml]: Done!

**Review by mauroLonghi — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**